IdentityCore/src/controllers/broker/MSIDBrokerExtensionInteractiveTokenRequestController.m

```diff
@@ -0,0 +1,87 @@
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "MSIDBrokerExtensionInteractiveTokenRequestController.h"
+#import "MSIDInteractiveRequestParameters.h"
+#import "MSIDInteractiveTokenRequest.h"
+
+@interface MSIDBrokerExtensionInteractiveTokenRequestController ()
+
+@property (nonatomic) MSIDInteractiveTokenRequest *currentRequest;
+
+@end
+
+@implementation MSIDBrokerExtensionInteractiveTokenRequestController
+
+- (instancetype)initWithInteractiveRequestParameters:(MSIDInteractiveRequestParameters *)parameters
+                                tokenRequestProvider:(id<MSIDTokenRequestProviding>)tokenRequestProvider
+                                  fallbackController:(id<MSIDRequestControlling>)fallbackController
+                                               error:(NSError **)error
+{
+    self = [super initWithRequestParameters:parameters
+                       tokenRequestProvider:tokenRequestProvider
+                         fallbackController:fallbackController
+                                      error:error];
+    if (self)
+    {
+        _interactiveRequestParameters = parameters;
+    }
+    
+    return self;
+}
+
+#pragma mark - MSIDRequestControlling
+
+- (void)acquireToken:(MSIDRequestCompletionBlock)completionBlock
+{
+    MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"Beginning interactive broker extension flow.");
+    
+    if (!completionBlock)
+    {
+        MSID_LOG_WITH_CTX(MSIDLogLevelError, self.requestParameters, @"Passed nil completionBlock. End interactive broker extension flow.");
+        return;
+    }
+    
+    self.currentRequest = [self.tokenRequestProvider interactiveBrokerExtensionTokenRequestWithParameters:self.interactiveRequestParameters];
+
+    [self.currentRequest executeRequestWithCompletion:^(MSIDTokenResult *result, NSError *error, __unused MSIDWebWPJResponse * msauthResponse)
+    {
+        MSIDRequestCompletionBlock completionBlockWrapper = ^(MSIDTokenResult * _Nullable result, NSError * _Nullable error)
+        {
+            MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"Interactive broker extension flow finished. Result %@, error: %ld error domain: %@", _PII_NULLIFY(result), (long)error.code, error.domain);
+            
+            self.currentRequest = nil;
+            completionBlock(result, error);
+        };
+        
+        completionBlockWrapper(result, error);
+    }];
+}
+
+- (BOOL)canPerformRequest
```

Comment:
previously we would check for presence of broker before initializing broker controller. Now, some of the checking is done before initialization and some after. Perhaps we should be consistent and always check for presence/capability of device before initialization a specific controller?

Reply:
Moved to static method.